tests/aws/services/sns/test_sns.py

```diff
@@ -3707,6 +3727,142 @@ def get_messages(_queue_url: str, _recv_messages: list):
             recv_messages.sort(key=itemgetter("Body"))
             snapshot.match(f"messages-queue-{i}", {"Messages": recv_messages})
 
+    @markers.aws.validated
+    def test_filter_policy_on_message_body_array_of_object_attributes(
+        self,
+        sqs_create_queue,
+        sns_create_topic,
+        sns_create_sqs_subscription_with_filter_policy,
+        snapshot,
+        aws_client,
+    ):
+        # example from https://aws.amazon.com/blogs/compute/introducing-payload-based-message-filtering-for-amazon-sns/
+        topic_arn = sns_create_topic()["TopicArn"]
+        queue_url = sqs_create_queue()
+        # complex filter policy with different level of nesting
+        filter_policy = {
+            "Records": {
+                "s3": {"object": {"key": [{"prefix": "auto-"}]}},
+                "eventName": [{"prefix": "ObjectCreated:"}],
+            }
+        }
+
+        sns_create_sqs_subscription_with_filter_policy(
+            topic_arn=topic_arn,
+            queue_url=queue_url,
+            filter_scope="MessageBody",
+            filter_policy=filter_policy,
+        )
+
+        response = aws_client.sqs.receive_message(
```

Comment:
I agree, that's maybe a bit overkill.. I think we had those very early in SNS tests (I guess because of bad cleanups), and it's something that stayed by.. tradition? I'll remove, this just takes space for nothing! Thanks for bringing this up! We sometimes forget to question things 😄